pkg/filesystem/local_directory_unix.go

```diff
@@ -24,6 +35,10 @@ func validateFilename(name string) error {
 	return nil
 }
 
+func PortableSyscallUmask(mask int) (oldmask int) {
```

Comment:
In this case we only call this function with zero and never use the return value. What if we were to call it this?

```go
func ClearUmask() {
	syscall.Umask(0)
}
```